packages/components/src/components/autocomplete/autocomplete.tsx

```diff
@@ -235,42 +262,35 @@ const Autocomplete = React.memo(props => {
         }
     };
 
-    const onSelectItem = item => {
+    const onSelectItem = (item: TItem) => {
         if (!item) return;
 
-        setInputValue(item.text ? item.text : item);
+        setInputValue((typeof item === 'object' ? item.text : item) || '');
 
-        if (typeof props.onItemSelection === 'function') {
-            props.onItemSelection(item);
-        }
+        props.onItemSelection?.(item);
     };
 
     const showDropdownList = () => {
         setShouldShowList(true);
 
-        if (typeof props.onShowDropdownList === 'function') {
-            props.onShowDropdownList();
-        }
+        props.onShowDropdownList?.();
     };
 
     const hideDropdownList = () => {
         setShouldShowList(false);
 
-        if (typeof props.onHideDropdownList === 'function') {
-            props.onHideDropdownList();
-        }
+        props.onHideDropdownList?.();
     };
 
-    const filterList = e => {
-        const val = e.target.value.toLowerCase();
+    const filterList = (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+        const val = e.currentTarget.value.toLowerCase();
```

Comment:
@jim-deriv why do we need to replace target with currentTarget here? )

Reply:
@maryia-deriv, because the `current` property apparently doesn't exist on  the  `FormEvent<HTMLInputElement | HTMLTextAreaElement>` type

Reply:
@jim-deriv https://github.com/niloofar-deriv/deriv-app/pull/90/files#r1133486778 😊